### ISSUE TITLE
3dvar on larger domain dies a horrible death

### ISSUE THREAD

**Issue by travissluka** [JCSDA-internal/soca#180]
something in the model is being repeatedly reinitialized during 3dvar. If you run `test_soca_3dvar_godas` you'll see that the following message is repeated for each PE on each iteration of the solver.  This doesn't crash the ctest, but on a single HPC node and 1 deg configuration eventually FMS dies because over a thousand netcdf files have been left open.

```
37: NOTE from PE     0: open_param_file: MOM_input has been opened successfully.
37:  MOM domain decomposition
37: whalo =    4, ehalo =    4, shalo =    4, nhalo =    4
37:   X-AXIS =   36  36
37:   Y-AXIS =   35
37:  MOMc domain decomposition
37: whalo =    2, ehalo =    2, shalo =    2, nhalo =    2
37:   X-AXIS =   18  18
37:   Y-AXIS =   17
37: NOTE from PE     0: close_param_file: MOM_input has been closed successfully.
```

**Comment by guillaumevernieres** [comment]
@travissluka , weird. I'm running the 3dvar on theia at 1/4 deg with no issues.

**Comment by guillaumevernieres** [comment]
@travissluka , @aerorahul : Looks like we had a snafu in one of the pr merges:
[soca_geom_mod.F90#L132](https://github.com/JCSDA/soca/blob/bcb59920949e687d75ecdcbe958961a3de2bf8b3/src/soca/Geometry/soca_geom_mod.F90#L132)

This isn't right, pointed out by @aerorahul a few days ago, I submitted a fix, but it's not in develop anymore. I suspect this could be the problem.

**Comment by aerorahul** [comment]
@guillaumevernieres your fix was absorbed into the open PR. May be we should just push this particular fix to develop, rather wait on the PR

**Comment by travissluka** [comment]
@aerorahul sounds good, I'll cherry-pick that one commit and do a separate PR for it

**Comment by aerorahul** [comment]
@travissluka look at L122 133,134 from PR #173.

**Comment by travissluka** [comment]
ah, nevermind, i see @guillaumevernieres  is on it....
